scripts/check_types.py

```diff
@@ -0,0 +1,104 @@
+"""
+Type hints checker using mypy.
+"""
+
+import json
+import logging
+import subprocess
+import sys
+import tempfile
+from collections import defaultdict
+from pathlib import Path
+
+# Set up logging
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+logger = logging.getLogger(__name__)
+
+
+def main():
+    # Setup
+    script_dir = Path(__file__).resolve().parent
+    dirs = sys.argv[1:] or [
+        script_dir.parent / "prolif",
+        script_dir.parent / "tests",
+    ]
+    dirs = [d for d in dirs if Path(d).exists()]
+    if not dirs:
+        logger.error("Error: No valid directories found")
+        return
+
+    # Find Python files
+    files = [
+        (d, Path(root) / f, str(Path(root) / f))
+        for d in dirs
+        for root, _, fs in subprocess.os.walk(d)
+        for f in fs
+        if f.endswith(".py")
+    ]
+
+    logger.info(f"Found {len(files)} Python files")
```

Comment:
Could you add a check that there's at least one file to run the type checking on (and log how many files in total) else raise an error?